src/app/donation-start/donation-start-form/donation-start-form.component.spec.ts

```diff
@@ -1,29 +1,102 @@
-import { RouterTestingModule } from '@angular/router/testing';
-import { RecaptchaModule } from 'ng-recaptcha';
-import { MatomoModule } from 'ngx-matomo';
-import { InMemoryStorageService } from 'ngx-webstorage-service';
-import { of } from 'rxjs';
-
-import { Campaign } from '../../campaign.model';
-import { TBG_DONATE_STORAGE } from '../../donation.service';
-import { TBG_DONATE_ID_STORAGE } from '../../identity.service';
-import { TimeLeftPipe } from "../../time-left.pipe";
+import {CurrencyPipe, DatePipe} from "@angular/common";
+import {HttpClientTestingModule} from '@angular/common/http/testing';
+import {ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
+import {FormBuilder, FormGroup, ReactiveFormsModule} from '@angular/forms';
+import {MatButtonModule} from '@angular/material/button';
+import {MatCheckboxModule} from '@angular/material/checkbox';
+import {MatDialog, MatDialogModule} from '@angular/material/dialog';
+import {MatIconModule} from '@angular/material/icon';
+import {MatInputModule} from '@angular/material/input';
+import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
+import {MatRadioModule} from '@angular/material/radio';
+import {MatSelectModule} from '@angular/material/select';
+import {MatStepperModule} from '@angular/material/stepper';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {ActivatedRoute, Router} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {RecaptchaModule} from 'ng-recaptcha';
+import {MatomoModule, MatomoTracker} from 'ngx-matomo';
+import {InMemoryStorageService} from 'ngx-webstorage-service';
+import {of} from 'rxjs';
+
+import {Campaign} from '../../campaign.model';
+import {DonationService, TBG_DONATE_STORAGE} from '../../donation.service';
+import {IdentityService, TBG_DONATE_ID_STORAGE} from '../../identity.service';
+import {TimeLeftPipe} from "../../time-left.pipe";
 import {DonationStartFormComponent} from "./donation-start-form.component";
+import {CardIconsService} from "../../card-icons.service";
+import {ChangeDetectorRef, ElementRef} from "@angular/core";
+import {ConversionTrackingService} from "../../conversionTracking.service";
+import {PageMetaService} from "../../page-meta.service";
+import {PostcodeService} from "../../postcode.service";
+import {StripeService} from "../../stripe.service";
+import {Donation} from "../../donation.model";
+
+function makeDonationStartFormComponent(donationService: DonationService,) {
+  const donationStartFormComponent = new DonationStartFormComponent(
+    undefined as unknown as CardIconsService,
+    undefined as unknown as ChangeDetectorRef,
+    {
+      convert: () => {
+      }
+    } as unknown as ConversionTrackingService,
+    undefined as unknown as MatDialog,
+    donationService,
+    undefined as unknown as ElementRef<any>,
+    undefined as unknown as FormBuilder,
+    undefined as unknown as IdentityService,
```

Comment:
Does this method of instantiation require all these @bdsl, or is it just a preference to make them explicitly `undefined`?

Reply:
I'm not sure I follow the question - the constructor of the component class requires all these arguments. 'as unknown as <type>' is just how we force the compiler to pretend that undefined is the required type. We know that for these particular tests those arguments are not used so we can just use undefined as a dummy value.

Reply:
Makes sense, I'd forgotten we had all these args in a constructor!